mobile/lib/shared/services/sync.service.dart

```diff
@@ -130,13 +138,46 @@ class SyncService {
     return true;
   }
 
-  /// Syncs remote assets to the databas
-  /// returns `true` if there were any changes
-  Future<bool> _syncRemoteAssetsToDb(
+  /// Efficiently syncs assets via changes. Returns `null` when a full sync is required.
+  Future<bool?> _syncRemoteAssetChanges(
```

Comment:
I believe this is the bread-and-butter of this PR, correct?

Can you help me walk through the logic in this method? Thank you

Reply:
Indeed! Sure, I'll describe the logic below:

- Get time of last successful sync from Isar DB
- ask server for changes (added/changed or deleted assets) since that last sync time
- in case the server can not correctly answer the request because the last sync is too long ago, return and require a full sync
- apply the changes to the Isar DB: try to delete deleted assets, update/merge/add changed remote assets
- if all went well, update the time of last successful sync to be just before the server was asked for changes